includes/Core/Permissions/Permissions.php

```diff
@@ -360,6 +360,21 @@ private function map_meta_capabilities( array $caps, $cap, $user_id, $args ) {
 			$caps = array_merge( $caps, $this->check_dashboard_sharing_capability( $cap, $user_id, $args ) );
 		}
 
+		// Handle VIEW_SPLASH capability.
+		if ( self::VIEW_SPLASH === $cap ) {
+			$caps = array_merge( $caps, $this->check_view_splash_capability( $user_id ) );
+		}
+
+		// Handle VIEW_AUTHENTICATED_DASHBOARD capability.
+		if ( self::VIEW_AUTHENTICATED_DASHBOARD === $cap ) {
+			$caps = array_merge( $caps, $this->check_view_authenticated_dashboard_capability( $user_id ) );
+		}
+
+		// Special Handling of VIEW_DASHBOARD and VIEW_POSTS_INSIGHTS capabilities when the dashboardSharing feature flag is enabled.
+		if ( in_array( $cap, array( self::VIEW_DASHBOARD, self::VIEW_POSTS_INSIGHTS ), true ) ) {
+			$caps = array_merge( $caps, $this->check_view_dashboard_capability( $user_id ) );
+		}
```

Comment:
How about using a `switch` here as well, similar to how `check_dashboard_sharing_capability`? The comments seem less necessary this way too.

<details>
<summary>diff</summary>

```diff
diff --git a/includes/Core/Permissions/Permissions.php b/includes/Core/Permissions/Permissions.php
index 8429863df7..b5504403e9 100644
--- a/includes/Core/Permissions/Permissions.php
+++ b/includes/Core/Permissions/Permissions.php
@@ -360,19 +360,18 @@ final class Permissions {
 			$caps = array_merge( $caps, $this->check_dashboard_sharing_capability( $cap, $user_id, $args ) );
 		}
 
-		// Handle VIEW_SPLASH capability.
-		if ( self::VIEW_SPLASH === $cap ) {
-			$caps = array_merge( $caps, $this->check_view_splash_capability( $user_id ) );
-		}
-
-		// Handle VIEW_AUTHENTICATED_DASHBOARD capability.
-		if ( self::VIEW_AUTHENTICATED_DASHBOARD === $cap ) {
-			$caps = array_merge( $caps, $this->check_view_authenticated_dashboard_capability( $user_id ) );
-		}
-
-		// Special Handling of VIEW_DASHBOARD and VIEW_POSTS_INSIGHTS capabilities when the dashboardSharing feature flag is enabled.
-		if ( in_array( $cap, array( self::VIEW_DASHBOARD, self::VIEW_POSTS_INSIGHTS ), true ) ) {
-			$caps = array_merge( $caps, $this->check_view_dashboard_capability( $user_id ) );
+		switch ( $cap ) {
+			case self::VIEW_SPLASH:
+				$caps = array_merge( $caps, $this->check_view_splash_capability( $user_id ) );
+				break;
+			case self::VIEW_AUTHENTICATED_DASHBOARD:
+				$caps = array_merge( $caps, $this->check_view_authenticated_dashboard_capability( $user_id ) );
+				break;
+			// Intentional fallthrough.
+			case self::VIEW_DASHBOARD:
+			case self::VIEW_POSTS_INSIGHTS:
+				$caps = array_merge( $caps, $this->check_view_dashboard_capability( $user_id ) );
+				break;
 		}
 
 		return $caps;
```

</details>